include/swift/SILOptimizer/Utils/PartitionUtils.h

```diff
@@ -299,10 +315,21 @@ class Partition {
       PartitionOp op,
       llvm::function_ref<void(const PartitionOp&, unsigned)>
       handleFailure = [](const PartitionOp&, unsigned) {},
-      std::vector<unsigned> nonconsumables = {},
+
+      std::vector<unsigned>
+          nonconsumables = {},
+
       llvm::function_ref<void(const PartitionOp&, unsigned)>
-      handleConsumeNonConsumable = [](const PartitionOp&, unsigned) {}
+      handleConsumeNonConsumable = [](const PartitionOp&, unsigned) {},
+
+      bool reviveAfterFailure = false
   ) {
+    auto handleFailureAndRevive =
```

Comment:
would be good to add a comment to explain the concept of "reviving" since it's a new concept / term. that is, how does the horizontal update "revive" and when is it valid to revive?